bundles/product-api-schedule-price-import/src/FondOfKudu/Zed/ProductApiSchedulePriceImport/Business/Model/SalePriceProductAbstractHandler.php

```diff
@@ -0,0 +1,148 @@
+     * @var \FondOfKudu\Zed\ProductApiSchedulePriceImport\ProductApiSchedulePriceImportConfig
+     */
+    protected ProductApiSchedulePriceImportConfig $apiSchedulePriceImportConfig;
+
+    /**
+     * @var \FondOfKudu\Zed\ProductApiSchedulePriceImport\Persistence\ProductApiSchedulePriceImportRepositoryInterface
+     */
+    protected ProductApiSchedulePriceImportRepositoryInterface $productApiSchedulePriceImportRepository;
+
+    /**
+     * @param \FondOfKudu\Zed\ProductApiSchedulePriceImport\Dependency\Facade\ProductApiSchedulePriceImportToPriceProductScheduleFacadeInterface $priceProductScheduleFacade
+     * @param \FondOfKudu\Zed\ProductApiSchedulePriceImport\Business\Mapper\PriceProductScheduleMapperInterface $priceProductScheduleMapper
+     * @param \FondOfKudu\Zed\ProductApiSchedulePriceImport\Persistence\ProductApiSchedulePriceImportRepositoryInterface $productApiSchedulePriceImportRepository
+     * @param \FondOfKudu\Zed\ProductApiSchedulePriceImport\ProductApiSchedulePriceImportConfig $apiSchedulePriceImportConfig
+     */
+    public function __construct(
+        ProductApiSchedulePriceImportToPriceProductScheduleFacadeInterface $priceProductScheduleFacade,
+        PriceProductScheduleMapperInterface $priceProductScheduleMapper,
+        ProductApiSchedulePriceImportRepositoryInterface $productApiSchedulePriceImportRepository,
+        ProductApiSchedulePriceImportConfig $apiSchedulePriceImportConfig
+    ) {
+        $this->priceProductScheduleFacade = $priceProductScheduleFacade;
+        $this->priceProductScheduleMapper = $priceProductScheduleMapper;
+        $this->apiSchedulePriceImportConfig = $apiSchedulePriceImportConfig;
+        $this->productApiSchedulePriceImportRepository = $productApiSchedulePriceImportRepository;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductAbstractTransfer $productAbstractTransfer
+     *
+     * @return \Generated\Shared\Transfer\ProductAbstractTransfer
+     */
+    public function handle(ProductAbstractTransfer $productAbstractTransfer): ProductAbstractTransfer
+    {
+        if (!$this->validateSpecialPriceAttributes($productAbstractTransfer)) {
+            return $productAbstractTransfer;
+        }
+
+        foreach ($productAbstractTransfer->getPrices() as $priceProductTransfer) {
+            $priceProductScheduleTransfer = $this->productApiSchedulePriceImportRepository
+                ->findPriceProductScheduleByIdProductAbstractAndIdCurrencyAndIdStore(
+                    $productAbstractTransfer->getIdProductAbstract(),
+                    $priceProductTransfer->getMoneyValue()->getFkCurrency(),
+                    $priceProductTransfer->getMoneyValue()->getFkStore(),
+                );
+
+            if ($priceProductScheduleTransfer === null) {
+                $this->create($productAbstractTransfer, $priceProductTransfer);
+            } else {
+                $this->update($productAbstractTransfer, $priceProductScheduleTransfer);
+            }
```

Comment:
Can we improve the branching ?.

Reply:
Ye I was thinking something like that it's just easier to read. Up to you though.